neuraxle/checkpoints.py

```diff
@@ -91,49 +105,89 @@ class PickleCheckpointStep(BaseCheckpointStep):
+                    pickle.load(file)
+                checkpoint_data_container.append(
+                    current_id=checkpoint_current_id,
+                    data_input=checkpoint_data_input,
+                    expected_output=checkpoint_expected_output
+                )
+
+        return checkpoint_data_container
 
-    def save_checkpoint(self, data_inputs):
+    def save_checkpoint(self, data_container: DataContainer) -> DataContainer:
         """
-        Save pickle files for data inputs and expected output
-        to create a checkpoint
-        :param data_inputs: data inputs to be saved in a pickle file
+        Save pickle files for data inputs and expected output to create a checkpoint
+
+        :param data_container: data to resume
         :return:
         """
-        # TODO: don't force the user to set the checkpoint name (use step name instead).
-        self.set_checkpoint_path(self.force_checkpoint_name)
-        with open(self.get_checkpoint_file_path(data_inputs), 'wb') as file:
-            pickle.dump(data_inputs, file)
+        for current_id, data_input, expected_output in data_container:
+            with open(self.get_checkpoint_file_path(current_id), 'wb') as file:
+                pickle.dump(
+                    (current_id, data_input, expected_output),
+                    file
+                )
+
+        return data_container
 
     def set_checkpoint_path(self, path):
         """
-        Set checkpoint path inside the cache folder (ex: cache_folder/pipeline_name/force_checkpoint_name/data_inputs.pickle)
+        Set checkpoint path inside the cache folder (ex: cache_folder/pipeline/step_a/current_id.pickle)
+
         :param path: checkpoint path
         """
         self.checkpoint_path = os.path.join(self.cache_folder, path)
         if not os.path.exists(self.checkpoint_path):
             os.makedirs(self.checkpoint_path)
 
-    def should_resume(self, data_inputs) -> bool:
-        return self.checkpoint_exists(data_inputs)
+    def should_resume(self, data_container: DataContainer) -> bool:
+        """
+        Whether or not we should resume the pipeline (if the checkpoint exists)
 
-    def checkpoint_exists(self, data_inputs) -> bool:
-        self.set_checkpoint_path(self.force_checkpoint_name)
-        return os.path.exists(self.get_checkpoint_file_path(data_inputs))
+        :param data_container: data to resume
+        :return:
+        """
+        return self._checkpoint_exists(data_container)
 
-    def get_checkpoint_file_path(self, data_inputs):
-        return os.path.join(self.checkpoint_path, 'data_inputs.pickle')
+    def _checkpoint_exists(self, data_container: DataContainer) -> bool:
+        """
+        Returns True if the checkpoints for each data input id exists
+        :param data_container:
+        :return:
+        """
+        for current_id, _, _ in data_container:
```

Comment:
Note that if for some reason, the iteration on the data_container returns a different number of things, you could do `for a, *_ in x:` (e.g.: if the data container is faked and doesn't contain any data_inputs values).